### DISCUSSION TITLE
PRICING_DECIMAL_PLACES messing currency formats

### DISCUSSION THREAD

**Discussion by anonkwe** [inventree/InvenTree#4498] [Q&A]
is there a way to globalize any format currency through django or python or anything ? We need to have the right decimals related to our locale and currency. but the option [PRICING_DECIMAL_PLACES] from the app is messing rendering a good amount of modules... is there a way to globalize it or bypass this setting at all?

**Comment by SchrodingersGat** [comment]
Please provide more information about what you are trying to do. Are you writing a report template? Or a plugin? What currency information are you currently trying to render?

**Comment by anonkwe** [comment]
it rounds up after the comma I think. decimal places are supposed to be
rendered after a "." in CAD currency. isn't there any way to just bypass
any config from the app itself. It just does not render the same anywhere
and it does not display null values in decimal places (ex: 1 instead of
1.00)

here is when it's set up at 2:

[image: image.png]

[image: image.png]

then, here is when it's at 3:
[image: image.png]

at 4:
[image: image.png]


here is an example from PO's when the value is at 5:

[image: image.png]

**Reply by anonkwe** [reply]
@SchrodingersGat anything tought of on how to render currency properly? any way to disable that inventree decimal places option in pricing settings, which breaks any django / python basic but correct and universal currency rendering. theres a lot of different places that decimals are rounded in the code as well... is there a way to bypass all this.. I think stock value and pricing is quite a core thing in such an app...

**Reply by SchrodingersGat** [reply]
You'll need to try to put those images up again. I cannot see what you are referring to as the images did not upload correctly

**Reply by anonkwe** [reply]
here is when it's set up at 2:
![image](https://user-images.githubusercontent.com/61838766/227408631-c3bd8cb9-4326-45b0-bfe2-2e8f8ca22f91.png)
![image](https://user-images.githubusercontent.com/61838766/227408690-6dfc5ddf-7c6b-4d1d-91a1-6f6581a3c406.png)

then, here is when it's at 3
![image](https://user-images.githubusercontent.com/61838766/227408717-481cdf2e-cca5-4ed4-9e4e-26fbcddf583a.png)

here's at 4
![image](https://user-images.githubusercontent.com/61838766/227408770-70a20d4d-6c29-4579-9d8d-5043643d18ae.png)

here is an example from PO's when the value is at 5:
![image](https://user-images.githubusercontent.com/61838766/227408809-be945105-c32f-4025-abfe-f339cdc1cf58.png)

**Comment by SchrodingersGat** [comment]
@anonkwe I think that the code is working as intended. I have checked different decimal places options locally, and in all the cases I checked it was working as I would expect.

The `PRICING_DECIMAL_PLACES` setting determines the *maximum* number of decimal places to render when displaying a currency. If there are *fewer* decimal places available, they are *not* rendered.

Examples below:

| Value | 3 Places | 5 Places |
| --- | --- | --- |
| 1.23456789 | 1.235 | 1.23457 |
| 4.44444444 | 4.444 | 4.44444 |
| 1.1111 | 1.111 | 1.1111 |
| 1.23 | 1.23 | 1.23 |
| 1.2000 | 1.2 | 1.2 |
| 7 | 7 | 7 |

Essentially, trailing zeroes are not rendered. Hope this clears things up?

**Comment by anonkwe** [comment]
currencies should roundup at 2 decimals %.2f (##,###.00) as it is by
default everywhere.. but there is no way to get this done partly because of
the pricing setting  and the bunch of rounded decimal values in the code

**Comment by SchrodingersGat** [comment]
I have raised a new issue for this - https://github.com/inventree/InvenTree/issues/4526

And a PR to fix it: https://github.com/inventree/InvenTree/pull/4527

You can also refer to the documentation to see how to render currency in a report:

https://docs.inventree.org/en/latest/report/helpers/#rendering-currency
